### ISSUE TITLE
Raspberry Pi Imager v 1.7.2: "Error writing file to disk"

### ISSUE THREAD

**Issue by dieseltravis** [raspberrypi/rpi-imager#473]
I'm trying to use Raspberry Pi Imager v1.7.2 on Windows 10 Pro, 64-bit 10.0.19044 to install [Home Assistant OS Installer for Yellow](https://yellow.home-assistant.io/power-supply-nvme/#installing-home-assistant-software-on-kit). After selecting the disk to write to, selecting Write, and then Yes, I get an error message "Error writing file to disk". I have tried with both a Generic 2GB and Sandisk 128GB USB drive (pictured below) and both give me the same error.

![image](https://user-images.githubusercontent.com/17863/185190149-0dc074e1-b667-4ddc-a38c-1d1366421743.png)
![image](https://user-images.githubusercontent.com/17863/185190179-af8d552d-2b0c-4e40-8e81-395658357bbc.png)
![image](https://user-images.githubusercontent.com/17863/185190251-ca1187a2-c1a0-459c-be25-ab3be68816ca.png)

**Comment by maxnet** [comment]
@agners
Could you compress the "Installer for Home Assistant Yellow Kit" image, so that it is a .img.xz/.img.gz/.zip file?

It is currently not compressed, and Imager has some issues on the Windows platform when streaming uncompressed files from the Internet.
(Imager tries to write the image data instantly in whatever chunk size it gets from the server while it downloads. But Windows only accepts writes that are a multiples of the 512 bytes sector size.
When dealing with compressed images, we have a different workflow that buffer things and this issue does not occur)

**Comment by agners** [comment]
@maxnet uh interesting! Sounds like a RPi Imager bug to me, but I can of course create a compressed image :smile: I'll have a look into it tomorrow.

Btw, we just today experienced what looks like https://github.com/raspberrypi/rpi-imager/issues/466. Could that be related to the fact that the image is uncompressed?

**Comment by maxnet** [comment]
>Sounds like a RPi Imager bug to me

It certainly is, and should be fixed.
But working around it is quicker than doing both that and convincing people to update to new version.

>Could that be related to the fact that the image is uncompressed?

Quite possible.
Recall Mac OS X does not like writes that are off sector size either.

(Linux however happily accepts writes in any size, so that's why it escaped testing).

**Comment by maxnet** [comment]
This build does support streaming uncompressed images from the Internet: [imager-1.7.3.exe](https://www.maxnet.eu/tmp/173-buffer/imager-1.7.3.exe)

**Comment by lurch** [comment]
> I can of course create a compressed image 

Given the amount of bandwidth and download-time it would save, I dunno why you would _want_ to stream an uncompressed image from the internet? :upside_down_face:

**Comment by dieseltravis** [comment]
@maxnet this works for me, thanks!

**Comment by agners** [comment]
> Given the amount of bandwidth and download-time it would save, I dunno why you would _want_ to stream an uncompressed image from the internet? :upside_down_face:

Well, what is faster depends heavily on the Internet speed. xz decompression is actually really not that fast, especially slow CPUs. I haven't done my own benchmark, but I've read 10MB/s on a RPi 4, so 100MBit Internet and faster makes uncompressed image actually the faster option. :sunglasses: 

But the main reason I opted for uncompressed are it's size, the image is very small (64MiB). And the fact that it uses a compressed squashfs inside (to boot the system in memory only, as this is an installer). So compressing the complete image again is somewhat weird.

In any case, I created a new release which also makes the image 1MB aligned, and xz compressed. Tests showed that this works now with MacOS systems.
